crates/bounded_planet_core/src/networking/events.rs

```
@@ -63,12 +66,109 @@ pub enum ReceiveEvent {
+        };
+
+        NetworkError::StreamSenderError {
+            connection: c,
+            stream: s,
+            failed_packet: f.clone()
+        }
+    }
+
+    pub async fn send(&self, senders: &mut Vec<SendStream<TlsSession>>, quinn_conn: &mut quinn::Connection) -> Result<(), NetworkError> {
+
+        match self {
+            SendEvent::SendPacket { stream, data, connection } => {
+                // Find (or create) the sender for this stream
+                let sender = match Self::get_stream_sender(&stream, senders, quinn_conn).await {
+                    Ok(sender) => sender,
+                    Err(err) => return Err(NetworkError::ConnectionError(err))
+                };
+
+                // Send the packet through the stream
+                if let Err(err) = data.send(sender).await {
+                    return Err(NetworkError::SendError {
+                        connection: *connection,
+                        stream: *stream,
+                        err,
+                    });
+                }
+
+                Ok(())
+            }
+
+            SendEvent::TransferPacket { data, .. } => {
+                // Open a new stream for this transfer
+                let uni = match quinn_conn.open_uni().await {
+                    Ok(uni) => uni,
+                    Err(err) => return Err(NetworkError::ConnectionError(err))
+                };
+
+                // Start sending the data
+                tokio::spawn(Self::send_transfer(uni, data.clone()));
+
+                Ok(())
+            }
+        }
+
+    }
+
+    async fn send_transfer(mut sender: SendStream<TlsSession>, data: Arc<Packet>) {
+        // Both of these method generate a result which is discarded.
+        // Results from transfers are not sent anywhere as that could potentially result in
+        // errors from a connection arriving in the ECS after is has closed!
+        let _ = data.send(&mut sender).await;
```

I can't tell if the comment is explaining why this is ok, or why it's a problem...

---

well... kinda both. I'm not entirely happy about these transfer errors being lost, but I'm also not sure it's a massive error as the client can just re-request what it needs if it never arrives.